### PR TITLE
Remove ---- delimiter for query error

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: sql/logic_test.go [hunk 1]
```diff
@@ -299,40 +299,50 @@ SET DATABASE = test;
...
 			for s.Scan() {
 				line := s.Text()
 				if line == "----" {
+					if query.expectErr != "" {
+						t.Fatalf("%s: invalid ---- line break after a query expecting an error: %s", query.pos, query.expectErr)
+					}
+					break
+				}
+				fields := strings.Fields(s.Text())
+				if len(fields) == 0 {
```
Comment (petermattis) [inline sql/logic_test.go]: No need to split into fields. This can be `if strings.TrimSpace(s.Text()) == ""`.

File: sql/logic_test.go [hunk 2]
```diff
@@ -299,40 +299,50 @@ SET DATABASE = test;
 					query.label = fields[3]
 				}
 			}
+
 			var buf bytes.Buffer
 			for s.Scan() {
 				line := s.Text()
 				if line == "----" {
+					if query.expectErr != "" {
+						t.Fatalf("%s: invalid ---- line break after a query expecting an error: %s", query.pos, query.expectErr)
```
Comment (petermattis) [inline sql/logic_test.go]: `s/line break/delimiter/g`?

**Comment by petermattis** [comment]
LGTM. You should double check that the sqllogictest files still run by `cd sql; make`. You don't have to let that complete, though, as it will take hours.

**Comment by vivekmenezes** [comment]
--- FAIL: TestLogic (157.95s)
	logic_test.go:490: ../../sqllogictest/test/index/between/10/slt_good_4.test:29631: expected:
		"1"  
		"5"  
		but found:
		

doesn't seem related.

**Comment by petermattis** [comment]
Yes, unrelated. I'll take a look at that failure.

**Comment by petermattis** [comment]
That test failure looks like what I fixed in #2416. Is it possible you didn't have that commit?

**Comment by petermattis** [comment]
Yes, your branch definitely didn't have #2416 so that explains the failure.

**Pull request merged**